### DISCUSSION TITLE
What's the best way to integrate a TCP server with a Pynecone backend?

### DISCUSSION THREAD

**Discussion by da4089** [reflex-dev/reflex#497] [Q&A]
I have an application that needs to have the backend server communicate with other services over a raw TCP protocol (ie. not WebSockets, not HTTP).

Could anyone offer a suggestion for the best way to integrate this with the Pynecone-generated backend application?

**Comment by Alek99** [comment]
Potentially with this section https://pynecone.io/docs/advanced-guide/api-routes you can have the backend communicate to your other services, it should be able to work with tcp it's using fast api under the hood not a networking person though

**Comment by da4089** [comment]
I can possibly use FastAPI's "startup" event to spawn a task into the asyncio event loop, and go from there, but I'm not sure how to get access to the right namespace to do that (or if it would actually even work).  I was just being lazy and hoping someone had already figured this out :-)

**Comment by da4089** [comment]
Thanks @Alek99!  You're right -- if I can't get anything else to work, it looks like I can add a REST API that way, and build the TCP service as a REST client of the backend, but ... that'd be a bit ugly, and I'd rather have everything in one process if there's a way to do it.

**Reply by Lendemor** [reply]
Not sure if it's a good solution for you, but you could look at what [RPyC](https://rpyc.readthedocs.io/en/latest/) does.

It use TCP for inter-process communication, so even though it's not "raw"-TCP (it has it's own service layer), it might give you some ideas.
